freeipa/src/main/java/com/sequenceiq/freeipa/nodestatus/NodeStatusJob.java

```diff
@@ -81,44 +76,41 @@ protected void executeTracedJob(JobExecutionContext context) throws JobExecution
 
     private void checkNodeHealthReports(Stack stack) {
         checkedMeasure(() -> {
-            ThreadBasedUserCrnProvider.doAsInternalActor(
-                    regionAwareInternalCrnGeneratorFactory.iam().getInternalCrnForServiceAsString(),
-                    () -> {
-                        if (unshedulableStates().contains(stack.getStackStatus().getStatus())) {
-                            LOGGER.debug("NodeStatus check will be unscheduled, stack state is {}", stack.getStackStatus().getStatus());
-                            nodeStatusJobService.unschedule(stack);
-                        } else {
-                            Set<InstanceMetaData> checkableInstances = stack.getAllInstanceMetaDataList().stream()
-                                    .filter(InstanceMetaData::isAvailable)
-                                    .collect(Collectors.toSet());
-                            if (checkableInstances.isEmpty()) {
-                                LOGGER.debug("No available instance for freeipa stack with id {}, skipping node status check.", stack.getId());
-                            } else {
-                                checkNodeHealthReports(stack, checkableInstances);
-                            }
-                        }
-                    });
+            if (unshedulableStates().contains(stack.getStackStatus().getStatus())) {
+                LOGGER.debug("NodeStatus check will be unscheduled, stack state is {}", stack.getStackStatus().getStatus());
+                nodeStatusJobService.unschedule(stack);
+            } else {
+                stack.getPrimaryGateway().ifPresent(primaryGateway -> {
+                    if (!primaryGateway.isAvailable()) {
+                        LOGGER.debug("No available instance for freeipa stack with id {}, skipping node status check.", stack.getId());
+                    } else {
+                        checkNodeHealthReports(stack, primaryGateway);
+                    }
```

Comment:
It would be nice to also add logging if the primary gateway is not present (e.g. using ifPresentOrElse instead of IfPresent), the previous code would also log in this case.

Reply:
Sure. I changed it.